src/init/starship.ps1

```diff
@@ -1,38 +1,79 @@
+        $_ = $_ -Replace '"','\"'         # Escape internal quote marks.
+        "`"$_`""                          # Quote the argument.
+    }
+    # Use the stop-parsing symbol (--%) to ensure the argument string is passed along unchanged 
+    # See: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing?view=powershell-5.1
+    Invoke-Expression "& `"$Executable`" --% $($Arguments -Join ' ')"
+}
+
 function global:prompt {
     $origDollarQuestion = $global:?
     $origLastExitCode = $global:LASTEXITCODE
 
-    $out = $null
     # @ makes sure the result is an array even if single or no values are returned
     $jobs = @(Get-Job | Where-Object { $_.State -eq 'Running' }).Count
-
-    $env:PWD = $PWD
-    $current_directory = (Convert-Path -LiteralPath $PWD)
     
-    # Whe start from the premise that the command executed correctly, which covers also the fresh console.
-    $lastExitCodeForPrompt = 0
+    $cwd = Get-Cwd
+    $arguments = @(
+        "prompt"
+        "--path=$($cwd.Path)",
+        "--logical-path=$($cwd.LogicalPath)",
+        "--jobs=$($jobs)"
+    )
 
     # Save old output encoding and set it to UTF-8
     $origOutputEncoding = [Console]::OutputEncoding
     [Console]::OutputEncoding = [System.Text.Encoding]::UTF8
+    
+    # Whe start from the premise that the command executed correctly, which covers also the fresh console.
+    $lastExitCodeForPrompt = 0
     if ($lastCmd = Get-History -Count 1) {
         # In case we have a False on the Dollar hook, we know there's an error.
         if (-not $origDollarQuestion) {
             # We retrieve the InvocationInfo from the most recent error.
             $lastCmdletError = try { Get-Error |  Where-Object { $_ -ne $null } | Select-Object -expand InvocationInfo } catch { $null }
-            # We check if the las command executed matches the line that caused the last error , in which case we know
+            # We check if the last command executed matches the line that caused the last error, in which case we know
             # it was an internal Powershell command, otherwise, there MUST be an error code.
             $lastExitCodeForPrompt = if ($null -ne $lastCmdletError -and $lastCmd.CommandLine -eq $lastCmdletError.Line) { 1 } else { $origLastExitCode }
         }
-
         $duration = [math]::Round(($lastCmd.EndExecutionTime - $lastCmd.StartExecutionTime).TotalMilliseconds)
-        # & ensures the path is interpreted as something to execute
-        $out = @(&::STARSHIP:: prompt "--path=$current_directory" --status=$lastExitCodeForPrompt --jobs=$jobs --cmd-duration=$duration)
-    } else {
-        $out = @(&::STARSHIP:: prompt "--path=$current_directory" --status=$lastExitCodeForPrompt --jobs=$jobs)
+        
+        $arguments += "--cmd-duration=$($duration)"
     }
+
+    $arguments += "--status=$($lastExitCodeForPrompt)"
+
+    $out = Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
```

Comment:
Just this is enough to get the prompt with the above changes:
`Invoke-Native -Executable ::STARSHIP:: -Arguments $argument`

Please get rid of `$out` and the `[Console]::OutputEncoding` stuff.